fightScreen.py

```diff
@@ -0,0 +1,633 @@
+import sys
+from random import choice as randChoice
+from time import sleep
+from PySide6.QtCore import QSize, Qt, QTimer
+from PySide6.QtGui import QPixmap, QMovie, QColor
+from PySide6.QtWidgets import (
+    QApplication,
+    QMainWindow,
+    QLabel,
+    QWidget,
+    QGridLayout,
+    QSpacerItem,
+    QSizePolicy,
+    QPushButton,
+)
+from meter import Meter
+from arsenal import Arsenal
+
+
+class MainWindow(QMainWindow):
+    def __init__(self):
+        super().__init__()
+
+        self.imageDir = "C:/classGame/"
+
+        self.userActionArray = []
+        self.compActionArray = []
+        self.imgDict = {
+            "sword": QPixmap(self.imageDir + "sword.jpg"),
+            "spear": QPixmap(self.imageDir + "spear.jpg"),
+            "shield": QPixmap(self.imageDir + "shield.jpg"),
+            "gloves": QPixmap(self.imageDir + "gloves.jpg"),
+        }
+        punch = QMovie(self.imageDir + "P1Attack1.gif")
+        kick = QMovie(self.imageDir + "P1Attack2.gif")
+        defense = QMovie(self.imageDir + "P1Defense.gif")
+        self.actionArray = ["Punch", "Kick", "Defend"]
+        self.damageArray = [10, 20, 0]
+        self.player1PicArray = [punch, kick, defense]
+        self.P1WeaponArray = {"sword": [20, 30, 10], "spear": [30, 10, 20]}
+        self.P1DefenseArray = {"shield": [30, 10, 20], "gloves": [10, 10, 10]}
+        self.P2WeaponArray = {"sword": [20, 30, 10], "spear": [30, 10, 20]}
+        self.P2DefenseArray = {"shield": [30, 10, 20], "gloves": [10, 10, 10]}
+        self.P1HealthMeter = 100
+        self.P2HealthMeter = 100
+        self.P1MagicMeter = 100
+        self.P2MagicMeter = 100
+        self.P1MechMeter = 100
+        self.P2MechMeter = 100
+        self.fightFlag = False
+        self.timer = QTimer()
+
+        self.setWindowTitle("My Fight App")
+
+        self.mainWidget = QWidget()
+        self.mainWidget.setStyleSheet("background-color: black;")
+        self.mainLayout = QGridLayout(spacing=0)
+        self.mainLayout.setContentsMargins(0, 0, 0, 0)
+        self.mainWidget.setLayout(self.mainLayout)
+
+        row = 0
+        colm = 0
+        self.mainLayout.addItem(
+            QSpacerItem(30, 50, QSizePolicy.Fixed, QSizePolicy.Fixed), row, 0
+        )
+        row += 1
+        colm += 1
+
+        #################################################################################
+        self.P1WeaponLayout = QGridLayout(spacing=0)
```

Comment:
Can we move this into it's own class or at least function? We could give it the #18 `player` as a parameter.
Same with the defense layout. See [here](https://github.com/chanson02/SENG3340/blob/c7290a0b1d6d0a3bde9d5ccf4ca7f98f4a473146/views/fight.py#L38).

Reply:
Actually let's do that with everything. This function is so long I can't get a good idea of what I'm looking at.

Reply:
I actually made one, but it wasn't showing in time before class so I put into main. I can upload this class draft after this and assign resolving errors as an issue.